source/CIS/cis-5.rst

```diff
@@ -0,0 +1,621 @@
+
+Each ``WithdrawCcdAmount`` is serialized as: the receiving address :ref:`CIS-2-Receiver` (``to``), the :ref:`CIS-5-CCDAmount` (``withdrawAmount``), and some additional data :ref:`CIS-2-AdditionalData` (``data``)::
+
+  WithdrawCcdAmount ::=  (to: Receiver) (withdrawAmount: CCDAmount) (data: AdditionalData)
+
+  WithdrawMessageCcdAmount ::= (entryPoint: EntrypointName) (expiryTime: TimeStamp) (nonce: u64) (serviceFeeRecipient: PublicKeyEd25519) (serviceFee: CCDAmount) (m: Byte²) (simpleWithdraws: WithdrawCcdAmountᵐ)
+
+  WithdrawBatchCcdAmount ::= (signer: PublicKeyEd25519) (signature: SignatureEd25519) (message: WithdrawMessageCcdAmount)
+
+  WithdrawParameterCcdAmount ::= (n: Byte²) (withdrawals: WithdrawBatchCcdAmountⁿ)
+
+.. _CIS-5-functions-transfer-ccd-receive-hook-parameter:
+
+CCD Receive hook parameter
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The parameter for the CCD receive hook function contains some additional data bytes.
+
+It is serialized as: some additional data :ref:`CIS-2-AdditionalData` (``data``)::
+
+  CCDReceiveHookParameter ::= (data: AdditionalData)
+
+Generating a valid signature
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To generate a valid signature for the entry point, the following bytes have to be signed::
+
+  WithdrawCCDSigningData ::= (chainContext: ChainContext) (param: WithdrawParameterCcdAmount)
+
+Requirements
+~~~~~~~~~~~~
+
+- The list of withdrawals MUST be executed in order.
+- The contract function MUST reject if any of the withdrawals fail to be executed.
+- The function MUST reject if the signature verification fails for any withdrawal.
+- The function MUST fail if the CCD balance of the ``signer`` is insufficient to do the withdrawal for any withdrawal.
+- A function MUST non-strictly decrease the CCD balance of the ``signer```s public key and non-strictly increase the balance of the ``to`` address or fail for any withdrawal.
+- A withdrawal back to this contract into the ``depositCcd`` entrypoint MUST be executed as a normal withdrawal.
+- A withdrawal of a CCD amount of zero MUST be executed as a normal withdrawal.
+- A withdrawal of any amount of CCD to a contract address MUST call a CCD receive hook function on the receiving smart contract with a :ref:`ccd receive hook parameter<CIS-5-functions-transfer-ccd-receive-hook-parameter>`.
+- The contract function MUST reject if the CCD receive hook function called on the contract receiving CCDs rejects for any withdrawal.
+- The balance of a public key not owning any CCD amount SHOULD be treated as having a balance of zero.
+- The function MUST transfer the ``serviceFee`` to the ``serviceFeeRecipient`` for each batch withdrawal if ``serviceFee!=0``.
+
+.. warning::
+
+  Be aware of transferring CCDs to a non-existing account address or contract address.
```

Comment:
I included the warning since this behavior is unusual/unexpected for most people coming from other ecosystems that an error can occur by sending CCD to a non-existing account address or contract address.

Reply:
I guess the warning is missing this then, i.e. how we behave differently.
The current message is not that informative